src/Microsoft.TemplateEngine.Cli/New3Command.cs

```diff
@@ -391,15 +391,39 @@ private CreationResultStatus EnterMaintenanceFlow()
                     Console.WriteLine();
                     Console.WriteLine(LocalizableStrings.InstalledItems);
 
-                    foreach (string value in _settingsLoader.InstallUnitDescriptorCache.InstalledItems.Values)
+                    foreach (KeyValuePair<Guid, string> entry in _settingsLoader.InstallUnitDescriptorCache.InstalledItems)
                     {
-                        Console.WriteLine($" {value}");
+                        Console.WriteLine($"  {entry.Value}");
 
-                        if (_settingsLoader.InstallUnitDescriptorCache.Descriptors.TryGetValue(value, out IInstallUnitDescriptor descriptor))
+                        if (_settingsLoader.InstallUnitDescriptorCache.Descriptors.TryGetValue(entry.Value, out IInstallUnitDescriptor descriptor))
                         {
                             if (descriptor.Details != null && descriptor.Details.TryGetValue("Version", out string versionValue))
                             {
-                                Console.WriteLine($"    {LocalizableStrings.Version}", versionValue);
+                                Console.WriteLine($"    {LocalizableStrings.Version} {versionValue}");
+                            }
+                        }
+
+                        HashSet<string> displayStrings = new HashSet<string>(StringComparer.Ordinal);
+
+                        foreach (TemplateInfo info in _settingsLoader.UserTemplateCache.TemplateInfo.Where(x => x.ConfigMountPointId == entry.Key))
```

Comment:
Nit: A mapping of the ConfigMountPointId's to their installed templates could be created before the outer loop, instead of iterating over all the templates for each of the install descriptors.

Reply:
A good point. This could save some time in cases where the template list is large.